Remove old board variants

mattytrentini opened the pull request:
I noticed a few unusual board variants and think they should be removed.

IDF3 builds are _very_ old now (it seems like the last successful builds are from 2021?). Can we remove them? If we want to retain the old builds for [ESP32_GENERIC](https://micropython.org/download/ESP32_GENERIC/) then we'll need to leave `ESP32_GENERIC/board.json` intact but we should at least remove it from `UM_TINYPICO/board.json` since there are [no IDF3 builds for that board](https://micropython.org/download/UM_TINYPICO/). 

If we remove IDF3 entirely we could also slightly simplify [build-boards.sh:61](https://github.com/micropython/micropython/blob/master/tools/autobuild/build-boards.sh#L61). I haven't made that change but can if we decide to remove IDF3.

Finally, I noticed a `wlan` variant for `ADAFRUIT_METRO_M4_EXPRESS`. I believe the samd port doesn't support variants so I think this is a mistake - and appears to be causing an unnecessary build.

dpgeorge commented:
Thanks, this looks good.

> IDF3 builds are _very_ old now (it seems like the last successful builds are from 2021?). Can we remove them?

Yes I think we should remove them.  If needed, one can build it themselves using IDF v3 (and this repo checked out at the appropriate commit in the far past).

> If we remove IDF3 entirely we could also slightly simplify [build-boards.sh:61](https://github.com/micropython/micropython/blob/master/tools/autobuild/build-boards.sh#L61). I haven't made that change but can if we decide to remove IDF3.

Yes, please do this as well.

Please can you split this PR into two commits:
- remove IDF v3 variants, and the bit from `build-boards.sh`
- remove the variant from `ADAFRUIT_METRO_M4_EXPRESS`

robert-hh commented:
There is a PR #12937 to add WLAN and BLE support for the SAMD51 family. The WLAN variant was created for that PR, and it seems to have sneaked into the master branch. But it harndly can be built, since the remainder of the PR is missing.

mattytrentini committed:
samd/boards/ADAFRUIT_METRO_M4_EXPRESS: Remove wlan variant.

There is no such variant.

Signed-off-by: Matt Trentini <matt.trentini@gmail.com>

mattytrentini commented:
This should be ready now @dpgeorge 👍🏻

dpgeorge commented on tools/autobuild/build-boards.sh:
```diff
@@ -55,10 +55,8 @@ function build_board {
-    # the idf3 files for older releases that used to be explicitly built).
-    for variant in `cat $board_json | python3 -c "import json,sys; print(' '.join(v for v in json.load(sys.stdin).get('variants', {}).keys() if v != 'IDF3'))"`; do
+    # Query variants from board.json and build them.
+    for variant in `cat $board_json | python3 -c "import json,sys; print(' '.join(v for v in json.load(sys.stdin).get('variants', {}).keys()))"`; do
```
This list comprehension is simple enough that the `v for v in` bit can now be removed.

mattytrentini replied:
Of course; fixed. 🤦🏻‍♂️

mattytrentini committed:
esp32/boards: Remove all IDF3 variants.

IDF 3 builds are very old now (it seems like the last successful builds are
from 2021), and the current IDF 5 is stable.  So remove IDF 3 variants.

Signed-off-by: Matt Trentini <matt.trentini@gmail.com>
